core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.spec.ts

```diff
@@ -74,4 +74,39 @@ describe('VisualizationPanelContentComponent', () => {
 
     expect(component.generateWordCloud).toHaveBeenCalled();
   });
+
+  it('should draw the spatial scatteplot map', () => {
+    const testData: Record<string, ResultObject> = {
+      'operator1': {
+        operatorID: 'operator1', chartType: ChartType.SPATIAL_SCATTERPLOT,
+        table: [{'xColumn': -90.285434, 'yColumn': 29.969126},
+          {'xColumn': -76.711521, 'yColumn': 39.197211}], totalRowCount: 2
+      }
+    };
+    spyOn(component, 'generateSpatialScatterplot');
+    spyOn(workflowStatusService, 'getCurrentResult').and.returnValue(testData);
+
+    component.operatorID = 'operator1';
+    component.ngAfterViewInit();
+
+    expect(component.generateSpatialScatterplot).toHaveBeenCalled();
```

Comment:
maybe we also want to assert that the HTML element is generated?

Reply:
If it's hard to make the HTML render in the test case (maybe because of no access token), then nevermind